File: javascript/backend-validator/02-country-follow-upward.mjs
// Set your CountriesDB private API key here
const API_KEY = 'your_private_api_key_here';

import { CountriesDBValidator } from '@countriesdb/validator';

if (!API_KEY || API_KEY === 'your_private_api_key_here') {
  throw new Error('Please set your COUNTRIESDB_PRIVATE_KEY in the script');
}

const validator = new CountriesDBValidator({
  apiKey: API_KEY,
});

// 2. Country validation with follow_upward
async function main() {
  try {
    console.log('Validating country code: US (with follow_upward)');
    const result = await validator.validateCountry('US', {
      followUpward: true,
    });
    console.log('Response:', JSON.stringify(result, null, 2));
    
    if (!result.valid) {
      throw new Error('Country validation failed');
    }
    
    console.log('Country is valid');
  } catch (err) {
    console.error('Validation failed:', err.message);
    process.exit(1);
  }
}

main();

File: python/backend-validator/02-country-follow-upward.py
"""Country validation with follow_upward example using countriesdb-validator package"""

from validator import CountriesDBValidator
import json
import sys

# Set your CountriesDB private API key here
API_KEY = 'your_private_api_key_here'

if not API_KEY or API_KEY == 'your_private_api_key_here':
    raise SystemExit('Please set your API_KEY in the script')

validator = CountriesDBValidator({
    'api_key': API_KEY,
})

# 2. Country validation with follow_upward
def main():
    try:
        print('Validating country code: US with follow_upward=True')
        result = validator.validate_country('US', follow_upward=True)
        print('Response:', json.dumps(result, indent=2))
        
        if not result['valid']:
            raise ValueError('Country validation failed')
        
        print('Country is valid')
    except Exception as exc:
        print(f'Validation failed: {exc}', file=sys.stderr)
        sys.exit(1)


if __name__ == '__main__':
    main()


